eng/scripts/Language-Settings.ps1

```diff
@@ -245,3 +245,178 @@ function GetExistingPackageVersions ($PackageName, $GroupId=$null)
+
+    $line = "$($item.Id),[$packageProperties]$($item.Name)"
+  } else { 
+    $line = "$($item.Id),$($item.Name)"
+  }
+
+  if ($item.Versions) {
+    $joinedVersions = $item.Versions -join ','
+    $line += ",$joinedVersions"
+  }
+
+  return $line
+}
+
+function Update-dotnet-DocsMsPackages($DocsRepoLocation, $DocsMetadata) {
+  UpdateDocsMsPackages `
+    (Join-Path $DocsRepoLocation 'bundlepackages/azure-dotnet-preview.csv') `
+    'preview' `
+    $DocsMetadata 
+
+  UpdateDocsMsPackages `
+    (Join-Path $DocsRepoLocation 'bundlepackages/azure-dotnet.csv') `
+    'latest' `
+    $DocsMetadata
+}
+
+function UpdateDocsMsPackages($DocConfigFile, $Mode, $DocsMetadata) {
+  Write-Host "Updating configuration: $DocConfigFile with mode: $Mode"
+  $packageConfig = Get-DocsCiConfig $DocConfigFile
+
+  $outputPackages = @()
+  foreach ($package in $packageConfig) {
+    # Do not filter by GA/Preview status because we want differentiate between
+    # tracked and non-tracked packages
+    $matchingPublishedPackageArray = $DocsMetadata.Where({ $_.Package -eq $package.Name })
+
+    # If this package does not match any published packages keep it in the list.
+    # This handles packages which are not tracked in metadata but still need to
+    # be built in Docs CI.
+    if ($matchingPublishedPackageArray.Count -eq 0) {
+      Write-Host "Keep non-tracked preview package: $($package.Name)"
+      $outputPackages += $package
+      continue
+    }
+
+    if ($matchingPublishedPackageArray.Count -gt 1) { 
+      LogWarning "Found more than one matching published package in metadata for $($package.Name); only updating first entry"
+    }
+    $matchingPublishedPackage = $matchingPublishedPackageArray[0]
+
+    if ($Mode -eq 'preview' -and !$matchingPublishedPackage.VersionPreview.Trim()) { 
+      # If we are in preview mode and the package does not have a superseding
+      # preview version, remove the package from the list. 
+      Write-Host "Remove superseded preview package: $($package.Name)"
+      continue
+    }
+
+    Write-Host "Keep tracked package: $($package.Name)"
+    $package.Versions = @($matchingPublishedPackage.VersionGA.Trim())
+    if ($Mode -eq 'preview') {
+      $package.Versions = @($matchingPublishedPackage.VersionPreview.Trim())
+    }
+    $outputPackages += $package
+  }
+
+  $outputPackagesHash = @{}
+  foreach ($package in $outputPackages) {
+    $outputPackagesHash[$package.Name] = $true
+  }
+
+  $remainingPackages = @() 
+  if ($Mode -eq 'preview') { 
+    $remainingPackages = $DocsMetadata.Where({ 
+      $_.VersionPreview.Trim() -and !$outputPackagesHash.ContainsKey($_.Package)
+    })
+  } else { 
+    $remainingPackages = $DocsMetadata.Where({ 
+      $_.VersionGA.Trim() -and !$outputPackagesHash.ContainsKey($_.Package)
+    })
+  }
+
+  # Add packages that exist in the metadata but are not onboarded in docs config
+  $newPackageProperties = [ordered]@{ tfm = 'netstandard2.0' }
```

Comment:
While this is the most common for our packages it is probably not a great thing to hard-code long term as we will be changing tfm or at least adding more tfm support to our packages. Probably worth filing a follow-up issue on this.

Reply:
https://github.com/Azure/azure-sdk-for-net/issues/22494